edx/edx-app-ios
# Customize course card image aspect ratio and iPad display

### Description

[OSPR-2117](https://openedx.atlassian.net/browse/OSPR-2117)

168c35af9a65ee98b84e210f50571d41aee8c49b Allows white-labeled apps to use a different aspect ratio for course card images by providing a different placeholder image.
The default placeholder image has aspect ratio 0.533, and so this change does not affect the default display.

edf9909474b1693982cb11c0f482cff2db755f8a Improves the course card image display on iPads by padding the course card table more when displaying on iPad.  iPhone padding is unchanged.

### Screenshots

* iPhone: 480x256 default placeholder image (default, no change)
  ![iphone 480x256 default](https://user-images.githubusercontent.com/7556571/33356312-b2cbe996-d50b-11e7-8b73-b328772258e5.png)
* iPhone: 378x226 placeholder image
  ![iphone 378x226](https://user-images.githubusercontent.com/7556571/33356115-a6a41d24-d50a-11e7-8dda-d7b3881af7ff.png)
* iPad: 480x256 default placeholder image (default)
  ![ipad default placeholder image](https://user-images.githubusercontent.com/7556571/36369378-eeda57b2-15aa-11e8-88a3-1c392bf428ae.png)
* iPad: 378 x 226 placeholder image
  ![ipad diff placeholder image](https://user-images.githubusercontent.com/7556571/36369682-11856ea4-15ac-11e8-8067-ff445ec985e2.png)

### Notes

* Using a different aspect ratio for the placeholder image can result in fewer or more course displayed on the screen than the standard 3 on iPhone.

### How to test this PR

1. Create a course with an image that uses a different aspect ratio than the standard 480x256.
1. View the course in the app on iPhone
  Note the change to the course card aspect ratio.
1. View the course in the app on iPad
  Note that the course images are not distorted, and are centred on the course card.

To use the sandbox server shown by the above screenshots, use:
```
API_HOST_URL: 'https://iospr1038.sandbox.opencraft.hosting'
OAUTH_CLIENT_ID: '73a0e26bc7df7cfeb707'
```

and login to the app as `honor@example.com` / `edx`

### Merge Deadline

Would like to get this merged before the next iOS app release.

### Reviewers

- [ ] @edx/mobile-product
